update few python3 packages

dkwo opened the pull request:
@ahesford Could you check whether the new jupyter client solves the issue with console?

ahesford commented:
I don't use jupyter; I wouldn't know how to test whether the console works properly or not.

dkwo commented:
There was this comment in `jupyter_client`
` # delay 6.2 until jupyter/jupyter_console#241 is fixed`
I thought you were affected by this issue.

dkwo commented:
Btw, all builds fine here, do not understand the entrypoints error.

dkwo commented:
Ok, works fine after adding a few depends.

dkwo commented:
I pushed a couple more simpe updates.
Any obstruction to merging this?

ahesford commented:
The `loop=loop` removal eliminates an obvious API incompatibility, but I suspect that there are some much deeper issues with how they handle the event loop that are just hidden by this change. I'm tempted to just leave `jupyter_console` untouched for now while they shake out the right approach.

paper42 reviewed (commented):
```
python3-soupsieve:
- soupsieve
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/soupsieve/__init__.py", line 29, in <module>
    from . import css_parser as cp
  File "/usr/lib/python3.10/site-packages/soupsieve/css_parser.py", line 5, in <module>
    from . import css_match as cm
  File "/usr/lib/python3.10/site-packages/soupsieve/css_match.py", line 7, in <module>
    import bs4  # type: ignore[import]
ModuleNotFoundError: No module named 'bs4'
```
```
python3-jupyter_client:
- jupyter_client
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/jupyter_client/__init__.py", line 11, in <module>
    from .manager import AsyncKernelManager  # noqa
  File "/usr/lib/python3.10/site-packages/jupyter_client/manager.py", line 32, in <module>
    from .provisioning import KernelProvisionerBase
  File "/usr/lib/python3.10/site-packages/jupyter_client/provisioning/__init__.py", line 1, in <module>
    from .factory import KernelProvisionerFactory  # noqa
  File "/usr/lib/python3.10/site-packages/jupyter_client/provisioning/factory.py", line 11, in <module>
    from entrypoints import EntryPoint  # type: ignore
ModuleNotFoundError: No module named 'entrypoints'
```
```
python3-httplib2:
- httplib2
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/httplib2/__init__.py", line 52, in <module>
    from . import auth
  File "/usr/lib/python3.10/site-packages/httplib2/auth.py", line 4, in <module>
    import pyparsing as pp
ModuleNotFoundError: No module named 'pyparsing'
```

dkwo commented:
Thanks. I believe those are now fixed. I removed any change to `jupyter_console`. (I will open an issue just to document present status, and not forget about it :)

dkwo commented:
see #35608

dkwo commented:
It seems they unbroke console, fixing #35608
Can this be merged?

paper42 commented on srcpkgs/python3-paramiko/template:
```diff
@@ -11,4 +11,6 @@ maintainer="Orphaned <orphan@voidlinux.org>"
 distfiles="${PYPI_SITE}/p/paramiko/paramiko-${version}.tar.gz"
-checksum=7f36f4ba2c0d81d219f4595e35f70d56cc94f9ac40a6acdf51d6ca210ce65035
+checksum=944a9e5dbdd413ab6c7951ea46b0ab40713235a9c4c5ca81cfe45c6f14fa677b
+# Testing via this command is deprecated
+make_check=no
```
same as above

dkwo replied:
This one I get
```
=> python3-paramiko-2.9.2_1: running do_check ...
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing manifest file 'paramiko.egg-info/SOURCES.txt'
running build_ext
Traceback (most recent call last):
  File "/builddir/paramiko-2.9.2/setup.py", line 51, in <module>
    setup(
  File "/usr/lib/python3.10/site-packages/setuptools/__init__.py", line 155, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 148, in setup
    return run_commands(dist)
  File "/usr/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
    dist.run_commands()
  File "/usr/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/usr/lib/python3.10/site-packages/setuptools/command/test.py", line 223, in run
    self.run_tests()
  File "/usr/lib/python3.10/site-packages/setuptools/command/test.py", line 226, in run_tests
    test = unittest.main(
  File "/usr/lib/python3.10/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python3.10/unittest/main.py", line 124, in parseArgs
    self._do_discovery(argv[2:])
  File "/usr/lib/python3.10/unittest/main.py", line 244, in _do_discovery
    self.createTests(from_discovery=True, Loader=Loader)
  File "/usr/lib/python3.10/unittest/main.py", line 154, in createTests
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/usr/lib/python3.10/unittest/loader.py", line 349, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/usr/lib/python3.10/unittest/loader.py", line 405, in _find_tests
    tests, should_recurse = self._find_test_path(
  File "/usr/lib/python3.10/unittest/loader.py", line 483, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/usr/lib/python3.10/site-packages/setuptools/command/test.py", line 56, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/usr/lib/python3.10/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/builddir/paramiko-2.9.2/paramiko/_winapi.py", line 55, in <module>
    class WindowsError(builtins.WindowsError):
AttributeError: module 'builtins' has no attribute 'WindowsError'
=> ERROR: python3-paramiko-2.9.2_1: do_check: 'python3 setup.py ${make_check_target} ${make_check_args}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:40
```

paper42 replied:
interesting, did you try removing the _winapi.py file?

dkwo replied:
now it wants pytest-relaxed, invoke

dkwo commented on srcpkgs/python3-jupyter_server/template:
```diff
@@ -14,7 +14,7 @@ maintainer="dkwo <nicolopiazzalunga@gmail.com>"
 distfiles="${PYPI_SITE}/j/jupyter-server/jupyter_server-${version}.tar.gz"
-checksum=5fb5a219385338b1d13a013a68f54688b6a69ecff4e757fd230e27ecacdbf212
+checksum=9e3e9717eea3bffab8cfb2ff330011be6c8bbd9cdae5b71cef169fcece2f19d3
 # Tarball defines no tests
 make_check=no
```
in which case, I get:
```
=> python3-jupyter_server-1.13.5_1: running do_check ...
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
/usr/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
  warnings.warn(
WARNING: The wheel package is not available.
/usr/bin/python3: No module named pip
error: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmprp9eg75u', '--quiet', 'argon2-cffi']' returned non-zero exit status 1.
=> ERROR: python3-jupyter_server-1.13.5_1: do_check: 'python3 setup.py ${make_check_target} ${make_check_args}' exited with 1
=> ERROR:   in do_check() at common/build-style/python3-module.sh:40
```

paper42 replied:
That's because argon2-cffi is in install_requires, but it's actually an optional dependency. We can either package it and add it to checkdepends or use make_check=no and add a comment `# needs python3-argon2-cffi which is not packaged`

dkwo replied:
after adding `python3-argon2`, as well as a ton of other `checkdepends`, it still fails with
```
=> python3-jupyter_server-1.13.5_1: running do_check ...
============================= test session starts ==============================
platform linux -- Python 3.10.3, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /builddir/jupyter_server-1.13.5, configfile: pyproject.toml, testpaths: jupyter_server/
plugins: mock-3.5.1, anyio-3.5.0, asyncio-0.14.0, timeout-1.4.2
timeout: 300.0s
timeout method: thread
timeout func_only: False
collected 0 items / 1 error

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
Defining 'pytest_plugins' in a non-top-level conftest is no longer supported:
It affects the entire test suite instead of just below the conftest as expected.
  /builddir/jupyter_server-1.13.5/jupyter_server/conftest.py
Please move it to a top level conftest file at the rootdir:
  /builddir/jupyter_server-1.13.5
For more information, visit:
  https://docs.pytest.org/en/stable/deprecations.html#pytest-plugins-in-non-top-level-conftest-files
=========================== short test summary info ============================
ERROR 
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.10s ===============================
```
What they actually require:
```
test =
    coverage
    pytest>=6.0
    pytest-cov
    pytest-mock
    pytest-timeout
    requests
    pytest-tornasync
    pytest-console-scripts
    ipykernel
```

dkwo replied:
ok, after updating, error message is clear: we are missing pytest_tornasync

dkwo commented on srcpkgs/python3-httplib2/template:
```diff
@@ -1,21 +1,23 @@
-checksum="8af66c1c52c7ffe1aa5dc4bcd7c769885254b0756e6e69f953c7f0ab49a70ba3
+checksum="58a98e45b4b1a48273073f905d2961666ecf0fbac4250ea5b47aef259eb5c585
  589eec38f72df2be203711d3b8cbece9b908c5e7ff00bc3cab7f63bae9e366b4"
+# Testing via this command is deprecated
+make_check=no
```
I've added several to checks, but it still fails one. It seems we are missing some checkdepends. Any ideas?

dkwo replied:
This only fails one test
```
FAILED tests/test_https.py::test_client_cert_password_verified - ValueError: ...
=================================== FAILURES ===================================
______________________ test_client_cert_password_verified ______________________
tests/test_https.py:151: in test_client_cert_password_verified
    expect_serial = tests.x509_serial(tests.CLIENT_ENCRYPTED_PEM) if tests.x509 else 16332984194609126128
        cert_log   = [{'issuer': ((('countryName', 'ZZ'),), (('stateOrProvinceName', '.'),), (('localityName', '.'),), (('organizationName'...tAfter': 'Oct  1 23:57:26 2031 GMT', 'notBefore': 'Oct  3 23:57:26 2021 GMT', 'serialNumber': 'E2AA6A96D1BF1AF0', ...}]
        handler    = <function test_client_cert_password_verified.<locals>.handler at 0x7f03cdb52440>
        http       = <httplib2.Http object at 0x7f03cdd9b670>
        setup_tls  = <function test_client_cert_password_verified.<locals>.setup_tls at 0x7f03cdb51d80>
        uri        = 'https://localhost:37437/'
        uri_parsed = ParseResult(scheme='https', netloc='localhost:37437', path='/', params='', query='', fragment='')
tests/__init__.py:786: in x509_serial
    cert = x509.load_pem_x509_certificate(pem)
        f          = <_io.BufferedReader name='/builddir/python3-httplib2-0.20.4/httplib2-0.20.4/tests/tls/client_encrypted.pem'>
        path       = '/builddir/python3-httplib2-0.20.4/httplib2-0.20.4/tests/tls/client_encrypted.pem'
        pem        = b'serial=E2AA6A96D1BF1AF0\nSHA1 Fingerprint=FB:42:16:53:02:8D:EA:F9:66:DB:BD:88:37:E1:93:7E:97:21:29:23\n-----BEGIN CE...LOQ4D+jK8/fcV9bhBuH\nLzBJ4gtPwUnvYqsaiIAeGi2EVllW0Ka+aTTzM1Yascl2q9WROvutAT0zz0M6smpO\n-----END RSA PRIVATE KEY-----\n'
/usr/lib/python3.10/site-packages/cryptography/x509/base.py:531: in load_pem_x509_certificate
    return rust_x509.load_pem_x509_certificate(data)
E   ValueError: Unable to load PEM file. See https://cryptography.io/en/latest/faq.html#why-can-t-i-import-my-pem-file for more details. InvalidData(InvalidByte(4, 45))
        backend    = None
        data       = b'serial=E2AA6A96D1BF1AF0\nSHA1 Fingerprint=FB:42:16:53:02:8D:EA:F9:66:DB:BD:88:37:E1:93:7E:97:21:29:23\n-----BEGIN CE...LOQ4D+jK8/fcV9bhBuH\nLzBJ4gtPwUnvYqsaiIAeGi2EVllW0Ka+aTTzM1Yascl2q9WROvutAT0zz0M6smpO\n-----END RSA PRIVATE KEY-----\n'
```
could you help me with this?

dkwo replied:
Left a comment, pointing to the issue.

paper42 commented on srcpkgs/python3-jupyter_server/template:
```diff
@@ -14,7 +14,7 @@ maintainer="dkwo <nicolopiazzalunga@gmail.com>"
 license="BSD-3-Clause"
 homepage="https://jupyter.org"
```
```suggestion
homepage="https://jupyter.org"
changelog="https://raw.githubusercontent.com/jupyter-server/jupyter_server/main/CHANGELOG.md"
```

dkwo replied:
Done.

dkwo commented on srcpkgs/python3-nbclassic/template:
```diff
@@ -11,7 +11,7 @@ maintainer="dkwo <nicolopiazzalunga@gmail.com>"
 distfiles="${PYPI_SITE}/n/nbclassic/nbclassic-${version}.tar.gz"
-checksum=99444dd63103af23c788d9b5172992f12caf8c3098dd5a35c787f0df31490c29
+checksum=7dbac0a6cb71bbe3afa1fe89369e6e3174d89f42aa08216a327046de45aa9a4f
 # Tarball includes no tests
 make_check=no
```
similarly, missing fixtures:
```
==================================== ERRORS ====================================
_____________________ ERROR at setup of test_tree_handler ______________________
file /builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py, line 19
  async def test_tree_handler(notebooks, jp_fetch):
      r = await jp_fetch('tree', 'nbclassic_test_notebooks')
      assert r.code == 200

      # Check that the tree template is loaded
      html = r.body.decode()
      assert "Files" in html
      assert "Running" in html
      assert "Clusters" in html
file /builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py, line 7
  @pytest.fixture
  def notebooks(jp_create_notebook):
E       fixture 'jp_create_notebook' not found
>       available fixtures: anyio_backend, anyio_backend_name, anyio_backend_options, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, notebooks, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, subtests, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py:7
___________________ ERROR at setup of test_notebook_handler ____________________
file /builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py, line 30
  async def test_notebook_handler(notebooks, jp_fetch):
      for nbpath in notebooks:
          r = await jp_fetch('notebooks', nbpath)
          assert r.code == 200
          # Check that the notebook template is loaded
          html = r.body.decode()
          assert "Menu" in html
          assert "Kernel" in html
          assert nbpath in html
file /builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py, line 7
  @pytest.fixture
  def notebooks(jp_create_notebook):
E       fixture 'jp_create_notebook' not found
>       available fixtures: anyio_backend, anyio_backend_name, anyio_backend_options, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, notebooks, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, subtests, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py:7
___________________ ERROR at setup of test_terminal_handler ____________________
file /builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py, line 41
  async def test_terminal_handler(jp_fetch):
          r = await jp_fetch('terminals', "1")
          assert r.code == 200
          # Check that the terminals template is loaded
          html = r.body.decode()
          assert "terminal-app" in html
E       fixture 'jp_fetch' not found
>       available fixtures: anyio_backend, anyio_backend_name, anyio_backend_options, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, notebooks, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, subtests, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/builddir/nbclassic-0.3.6/nbclassic/tests/test_notebookapp.py:41
=========================== short test summary info ============================
ERROR nbclassic/tests/test_notebookapp.py::test_tree_handler
ERROR nbclassic/tests/test_notebookapp.py::test_notebook_handler
ERROR nbclassic/tests/test_notebookapp.py::test_terminal_handler
```

dkwo commented:
I've added comments as to what is missing.
I feel httplib2 is close to passing tests, but I do not understand the only failure remaining.

dkwo commented:
I think the last test failure left with httplib2 is due to our cryptography being too recent (they require 35.0.0).
I opened an issue upstream, and left a comment in our template.

dkwo commented:
When trying to update some of these packages,
I found that they fail to build, unless I switch
to build them as python3-pep517. Is this fine?

dkwo commented:
Now I have troubles updating `tornado`, which is required for ipython: would it be ok to skip tests for it?

ahesford commented:
Skipping tests that previously succeeded but now fail requires a good explanation.

if the tests always failed an we just overlooked them, a good explanation of the failure would still be appreciated, but a justification of nothing more than `# FIXME` would be acceptable.

dkwo commented:
Now I'm stuck with `python3-ipython_ipykernel` update: why does it complain
```
=> python3-ipython_ipykernel-6.17.1_1: running do_build ...
* Getting build dependencies for wheel...

ERROR Missing dependencies:
	jupyter_client>=6
	jupyter-core>=4.9.2 -> platformdirs
=> ERROR: python3-ipython_ipykernel-6.17.1_1: do_build: 'python3 -m build ${make_build_args} ${make_build_target}' exited with 1
=> ERROR:   in do_build() at common/build-style/python3-pep517.sh:8
```
even though jupyter_client is in hostmakedeps?

dkwo commented:
@ahesford I finally figured it out, can you take a look?

ahesford commented on srcpkgs/python3-httplib2/template:
```diff
@@ -1,26 +1,29 @@
+distfiles="${PYPI_SITE}/h/httplib2/httplib2-${version}.tar.gz"
+checksum=fc144f091c7286b82bec71bdbd9b27323ba709cc612568d3000893bfd9cb4b34
+
+do_check() {
+	# https://github.com/httplib2/httplib2/issues/221
+	PYTHONPATH="$PWD/build/lib" python3 -m pytest tests/ -k 'not test_client_cert_password_verified'
+}
```
```suggestion
make_check_pre="PYTHONPATH=\${PWD}/build/lib"
make_check_args="-k not(test_client_cert_password_verified)"
```
is should work just as well, although you might need `make_check_target="tests"`.

ahesford replied:
Welp I fired from the hip here. I believe you can drop `make_check_pre` altogether and just use `make_check_args`.

dkwo replied:
Done.

ahesford commented on srcpkgs/python3-ipython_ipykernel/template:
```diff
@@ -1,19 +1,24 @@
+checksum=1ab68d3d3654196266baa93990055413e167263ffbe4cfe834f871bcd3d3506d
+make_check=no # cycle with python3-jupyter_core and python3-jupyter_client
+
+do_install() {
+	python3 -m installer --destdir ${DESTDIR} ${make_install_args} ${make_install_target} dist/*.whl
+}
```
```suggestion
make_install_target="dist/ipython_ipykernel-${version}-*-*-*.whl"
```

ahesford replied:
Another untested suggestion. The actual value for `make_install_target` should be

```bash
make_install_target="dist/ipykernel-${version}-*-*-*.whl"
```

Sorry for the noise....

dkwo replied:
done.

tornaria commented:
Something bad happened here:
```
$ sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 9.7, Release Date: 2022-09-19                     │
│ Using Python 3.11.1. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
/usr/lib/python3.11/site-packages/jupyter_client/__init__.py:23: UserWarning: Could not import submodules
  warnings.warn("Could not import submodules")
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3.11/pathlib.py", line 1250, in is_dir
    return S_ISDIR(self.stat().st_mode)
                   ^^^^^^^^^
AttributeError: 'str' object has no attribute 'stat'

Original exception was:
Traceback (most recent call last):
  File "/usr/lib/sagemath/9.7/bin/sage-ipython", line 15, in <module>
    app.initialize()
  File "/usr/lib/python3.11/site-packages/traitlets/config/application.py", line 110, in inner
    return method(app, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/IPython/terminal/ipapp.py", line 279, in initialize
    self.init_shell()
  File "/usr/lib/python3.11/site-packages/sage/repl/interpreter.py", line 765, in init_shell
    self.shell = self.shell_class.instance(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/traitlets/config/configurable.py", line 551, in instance
    inst = cls(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/IPython/terminal/interactiveshell.py", line 651, in __init__
    super(TerminalInteractiveShell, self).__init__(*args, **kwargs)
  File "/usr/lib/python3.11/site-packages/IPython/core/interactiveshell.py", line 610, in __init__
    self.init_display_formatter()
  File "/usr/lib/python3.11/site-packages/sage/repl/interpreter.py", line 288, in init_display_formatter
    backend.get_display_manager().switch_backend(backend, shell=self)
  File "/usr/lib/python3.11/site-packages/sage/repl/rich_output/display_manager.py", line 340, in switch_backend
    self._backend.install(**kwds)
  File "/usr/lib/python3.11/site-packages/sage/repl/rich_output/backend_ipython.py", line 60, in install
    from sage.repl.display.formatter import SageDisplayFormatter
  File "/usr/lib/python3.11/site-packages/sage/repl/display/formatter.py", line 69, in <module>
    from ipywidgets import Widget
  File "/usr/lib/python3.11/site-packages/ipywidgets/__init__.py", line 25, in <module>
    from .widgets import *
  File "/usr/lib/python3.11/site-packages/ipywidgets/widgets/__init__.py", line 4, in <module>
    from .widget import Widget, CallbackDispatcher, register, widget_serialization
  File "/usr/lib/python3.11/site-packages/ipywidgets/widgets/widget.py", line 18, in <module>
    from ipykernel.comm import Comm
  File "/usr/lib/python3.11/site-packages/ipykernel/__init__.py", line 5, in <module>
    from .connect import *  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/ipykernel/connect.py", line 12, in <module>
    from jupyter_client import write_connection_file
ImportError: cannot import name 'write_connection_file' from 'jupyter_client' (/usr/lib/python3.11/site-packages/jupyter_client/__init__.py)
```
@dkwo @ahesford: sage was working 5 minutes ago before I updated my system... Something in the list below broke it:
```
Name                      Action    Version           New version            Download size
libplacebo                update    4.192.1_1         4.208.0_1              291KB 
mpv                       update    0.35.0_1          0.35.0_2               1668KB 
python3-defusedxml        update    0.6.0_6           0.7.1_1                21KB 
python3-comm              install   -                 0.1.1_1                6439B 
python3-ipython_ipykernel update    5.5.4_3           6.19.2_1               105KB 
python3-jupyter_client    update    6.1.12_3          7.4.5_1                98KB 
python3-jupyter_console   update    6.4.0_3           6.4.4_1                21KB 
python3-jupyter_core      update    4.7.1_3           5.0.0_1                50KB 
python3-nbclient          update    0.5.10_2          0.7.0_1                60KB 
python3-soupsieve         update    1.9.5_9           2.3.2_1                34KB 
python3-terminado         update    0.9.4_3           0.17.0_1               15KB 
python3-tornado           update    6.1_4             6.2_1                  317KB 
python3-traitlets         update    5.0.5_3           5.5.0_1                91KB 
```

tornaria commented:
In fact, jupyter itself looks broken:
```
$ jupyter
Traceback (most recent call last):
  File "/bin/jupyter", line 5, in <module>
    from jupyter_core.command import main
  File "/usr/lib/python3.11/site-packages/jupyter_core/command.py", line 20, in <module>
    from . import paths
  File "/usr/lib/python3.11/site-packages/jupyter_core/paths.py", line 22, in <module>
    import platformdirs
ModuleNotFoundError: No module named 'platformdirs'
```

dkwo commented:
@tornaria installing python3-platformdirs fixes it

dkwo commented:
see https://github.com/void-linux/void-packages/pull/41046
